tapeagents/agent.py

```diff
@@ -686,13 +687,11 @@ def _run_implementation():
                     else:
                         raise ValueError("Agent can only generate steps or partial steps")
                 n_iterations += 1
-            updated_metadata = tape.metadata.model_copy(
-                update=dict(
-                    parent_id=input_tape_id,
-                    author=self.name,
-                    n_added_steps=len(tape) - input_tape_length,
-                )
-            )
+            updated_metadata = original_metadata.model_validate(dict(
```

Comment:
Are you sure that this will not erase the content of the original metadata?

Reply:
Do we even want to copy the metadata though? We have `tape.context` that is copied when the tape is continued.

With this code I tried to create new metadata but of the same type.

Reply:
We're using metadata to store the original task and the result in Gaia, that way the original task's dict will be lost and we cannot run 'tape_correct(tape)' anymore, which compares the golden result from metadata with the achieved.

Reply:
But we can move the addition of the task dict to the end of the 'solve_task()', after the orchestrator loop. But I would say it was convenient to maintain metadata from the tape that we're started with